vault/src/main.rs

```diff
@@ -48,7 +48,7 @@ where
     let blocking_task = tokio::task::spawn(future);
     tokio::select! {
         res = blocking_task => {
-            let _ = res?;
+            return res?;
```

Comment:
Is the type of `res` `Result<Result<..>, ..>`? How does that work?

Reply:
Yes because the `spawn` wraps it in another result.